en/lessons/advanced/nerves.md

```diff
@@ -0,0 +1,262 @@
+
+If the network has been functional after the previous step, that's it!
+
+You can finally stop switching SD cards: `mix firmware && ./upload.sh 192.168.88.2` is a combination of two commands: first creates a firmware, and the second one burns it via the network and reboots the device.
+
+_Hint: `ssh 192.168.88.2` gives you a proper IEx shell on the device in the context of the app. _
+
+_Troubleshooting: If you don't have an existing ssh key in your home folder, you will have an error `No SSH public keys found in ~/.ssh.`. In this case, you will need to run `ssh-keygen` and re-burn the firmware to use the network update feature._
+
+## Setting up the LED control
+
+To interact with LEDs, you need [nerves_leds](https://github.com/nerves-project/nerves_leds) package installed which, as you know, is done by adding `{:nerves_leds, "~> 0.8", targets: @all_targets},` to `mix.exs` file.
+
+After setting up the dependency, you need to configure the LED list for the given device. For example, for all Raspberry Pi models, there is only one LED onboard: `led0`. Let's use it by adding a `config :nerves_leds, names: [green: "led0"]` line to the `config/config.exs`.
+
+For other devices, you can take a look at the [corresponding part of the nerves_examples project](https://github.com/nerves-project/nerves_examples/tree/master/hello_leds/config).
+
+After configuring the LED itself, we surely need to control it somehow. To do that, we will add a GenServer in `lib/network_led/blinker.ex` with these contents:
+
+```elixir
+defmodule NetworkLed.Blinker do
+  use GenServer
+
+  @moduledoc """
+    Simple GenServer to control GPIO #18.
+  """
+
+  require Logger
+  alias Nerves.Leds
+
+  def start_link(state \\ []) do
+    GenServer.start_link(__MODULE__, state, name: __MODULE__)
+  end
+
+  def init(state) do
+    enable()
+
+    {:ok, state}
+  end
+
+  def handle_cast(:enable, state) do
+    Logger.info("Enabling LED")
+    Leds.set(green: true)
+
+    {:noreply, state}
+  end
+
+  def handle_cast(:disable, state) do
+    Logger.info("Disabling LED")
+    Leds.set(green: false)
+
+    {:noreply, state}
+  end
+
+  def enable() do
+    GenServer.cast(__MODULE__, :enable)
+  end
+
+  def disable() do
+    GenServer.cast(__MODULE__, :disable)
+  end
+end
+
+```
+
+To enable this, you also need to add it to the supervision tree in `lib/network_led/application.ex`: add `{NetworkLed.Blinker, name: NetworkLed.Blinker}` under the `  def children(_target) do` group.
+
+Notice that Nerves has two different supervision trees in application - one for the host machine and one for actual devices.
+
+After this - that's it! You actually can upload the firmware and check that `NetworkLed.Blinker.disable()` disables LED (which is enabled by default in code), and `NetworkLed.Blinker.enable()` enables it.
```

Comment:
How do I check? IEx on the host, or on the target device? Worth just noting.

Reply:
Thanks for noting, added.